src/sqlfluff/dialects/dialect_tsql.py

```diff
@@ -155,6 +345,79 @@ class DatatypeSegment(BaseSegment):
     )
 
 
+@tsql_dialect.segment()
+class NextValueSequenceSegment(BaseSegment):
+    """Segment to get next value from a sequence."""
+
+    type = "sequence_next_value"
+    match_grammar = Sequence(
+        "NEXT",
+        "VALUE",
+        "FOR",
+        Ref("ObjectReferenceSegment"),
+    )
+
+
+@tsql_dialect.segment()
+class IfExpressionStatement(BaseSegment):
+    """IF-ELSE-END IF statement.
+
+    https://docs.microsoft.com/en-us/sql/t-sql/language-elements/if-else-transact-sql?view=sql-server-ver15
+    """
+
+    type = "if_then_statement"
+
+    match_grammar = Sequence(
+        OneOf(
+            Sequence(Ref("IfNotExistsGrammar"), Ref("SelectStatementSegment")),
+            Sequence(Ref("IfExistsGrammar"), Ref("SelectStatementSegment")),
+            "IF",
+            Ref("ExpressionSegment"),
+        ),
+        Ref("StatementSegment"),
+        Sequence("ELSE", Ref("StatementSegment"), optional=True),
```

Comment:
Can we get a test case for ELSE?  I couldn't get ELSE to parse correctly with a similar approach.

Reply:
If else with begin end etc is still pending. Was also not able to get it working yet.